entity-resolution-fraud-detection/Communities_in_P2P_Quickstart_with_Louvain_and_PageRank.ipynb

```diff
@@ -0,0 +1,251 @@
+  },
+  {
+   "cell_type": "code",
+   "id": "96c76177-7e01-484a-9292-f3fc3f0e5a9d",
+   "metadata": {
+    "language": "sql",
+    "name": "roles",
+    "collapsed": false,
+    "codeCollapsed": false
+   },
+   "outputs": [],
+   "source": "-- you must be accountadmin to create role and grant permissions\nuse role accountadmin;",
+   "execution_count": null
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f9e24c11-30cc-46b1-bf78-a48638c45dbd",
+   "metadata": {
+    "name": "step2",
+    "collapsed": false
+   },
+   "source": "Next let's set up the necessary roles, permissions, and resource access to enable Graph Analytics to operate on data within the `p2p_demo.public` schema. It creates a consumer role (`gds_role`) for users and administrators, grants the Graph Analytics application access to read from and write to tables and views, and ensures that future tables are accessible. \n\nIt also provides the application with access to the required compute pool and warehouse resources needed to run graph algorithms at scale."
+  },
+  {
+   "cell_type": "code",
+   "id": "88f80353-e278-4fca-bb75-af3e76a2ab43",
+   "metadata": {
+    "language": "sql",
+    "name": "dbaccess"
+   },
+   "outputs": [],
+   "source": "USE SCHEMA P2P_DEMO.PUBLIC;\n\n-- Create a consumer role for users and admins of the GDS application\nCREATE ROLE IF NOT EXISTS gds_user_role;\nCREATE ROLE IF NOT EXISTS gds_admin_role;\nGRANT APPLICATION ROLE neo4j_graph_analytics.app_user TO ROLE gds_user_role;\nGRANT APPLICATION ROLE neo4j_graph_analytics.app_admin TO ROLE gds_admin_role;\n\nCREATE DATABASE ROLE IF NOT EXISTS gds_db_role;\nGRANT DATABASE ROLE gds_db_role TO ROLE gds_user_role;\nGRANT DATABASE ROLE gds_db_role TO APPLICATION neo4j_graph_analytics;\n\n-- Grant access to consumer data\nGRANT USAGE ON DATABASE P2P_DEMO TO ROLE gds_user_role;\nGRANT USAGE ON SCHEMA P2P_DEMO.PUBLIC TO ROLE gds_user_role;\n\n-- Required to read tabular data into a graph\nGRANT SELECT ON ALL TABLES IN DATABASE P2P_DEMO TO DATABASE ROLE gds_db_role;\n\n-- Ensure the consumer role has access to created tables/views\nGRANT ALL PRIVILEGES ON FUTURE TABLES IN SCHEMA P2P_DEMO.PUBLIC TO DATABASE ROLE gds_db_role;\nGRANT ALL PRIVILEGES ON ALL TABLES IN SCHEMA P2P_DEMO.PUBLIC TO DATABASE ROLE gds_db_role;\nGRANT CREATE TABLE ON SCHEMA P2P_DEMO.PUBLIC TO DATABASE ROLE gds_db_role;\nGRANT CREATE VIEW ON SCHEMA P2P_DEMO.PUBLIC TO DATABASE ROLE gds_db_role;\nGRANT ALL PRIVILEGES ON FUTURE VIEWS IN SCHEMA P2P_DEMO.PUBLIC TO DATABASE ROLE gds_db_role;\nGRANT ALL PRIVILEGES ON ALL VIEWS IN SCHEMA P2P_DEMO.PUBLIC TO DATABASE ROLE gds_db_role;\n\n-- Compute and warehouse access\nGRANT USAGE ON WAREHOUSE GDSONSNOWFLAKE TO APPLICATION neo4j_graph_analytics;\n",
```

Comment:
Please avoid usage of `GDS` acronym, it is "Neo4j Graph Analytics application" or "the application"

There are also some places with GDS acronym, e.g. GDSONSNOWFLAKE warehouse name

Reply:
Working on removing GDS.
Changed warehouse name in all files to `NEO4J_GRAPH_ANALYTICS_APP_WAREHOUSE`